File: non_native/lib/repositories/i_flashcards_repository.dart
import '../models/flashcard.dart';

abstract class IFlashcardsRepository {
  Stream<List<Flashcard>> getAllFlashcards(int categoryId);

  Future<Flashcard?> getFlashcardById(int id);
  Future<Flashcard> addFlashcard(Flashcard flashcard);
  Future<void> deleteFlashcard(int id);
  Future<void> updateFlashcard(Flashcard flashcard);
}

File: Native/src/main/java/com/example/mindflip_native/repositories/IFlashcardsRepository.kt
package com.example.mindflip_native.repositories

import com.example.mindflip_native.models.Flashcard

interface IFlashcardsRepository {
    fun getAllFlashcards(categoryId: Int): List<Flashcard>
    fun getFlashcardById(id: Int): Flashcard?
    fun addFlashcard(flashcard: Flashcard): Flashcard
    fun deleteFlashcard(id: Int)
    fun updateFlashcard(flashcard: Flashcard): Flashcard
}